### PR TITLE
Allow lint command to parse pre-release charts

### LINKED ISSUE #5150: 'helm lint' fails on chart packaged with version from output of 'git describe'
Given a Git tag `1.0.0`, a commit immediately following the tag is formatted something like `1.0.0-1-g7627627` when calling `git describe <COMMIT>`. If that version, which is a semantic version, is used as the version in a chart's `Chart.yaml` file, `helm lint` executes as expected. If the chart is then packaged via `helm package` or a chart with a different version number in its `Chart.yaml` file is packaged via `helm package --version 1.0.0-1-g7627627`, calling `helm lint` fails with the following error:

```bash
$ helm lint my-chart-1.0.0-1-g7627627.tgz
==> Skipping my-chart-1.0.0-1-g7627627.tgz
No chart found for linting (missing Chart.yaml)

Error: 0 chart(s) linted, 1 chart(s) failed
```

Inspecting the `*.tgz` file, however, reveals that `Chart.yaml` is present:

```bash
$ tar -tf my-chart-1.0.0-1-g7627627.tgz
my-chart/Chart.yaml
my-chart/values.yaml
my-chart/templates/NOTES.txt
my-chart/templates/_helpers.tpl
my-chart/templates/deployment.yaml
my-chart/templates/ingress.yaml
my-chart/templates/service.yaml
my-chart/.helmignore
```

Unpacking the `*.tgz` file and calling `helm lint` also passes, though this is likely expected given that `helm lint` also passes when the version is present in the exploded chart's `Chart.yaml` file.

Though `helm lint --help` doesn't explicitly say that it supports packaged charts, linting the exact same chart works when it has been packaged via `helm package --version 1.0.0`

Output of `helm version`:

```bash
$ helm version
Client: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.11.0", GitCommit:"2e55dbe1fdb5fdb96b75ff144a339489417b146b", GitTreeState:"clean"}
```

Output of `kubectl version`:

```bash
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.2", GitCommit:"17c77c7898218073f14c8d573582e8d2313dc740", GitTreeState:"clean", BuildDate:"2018-10-30T21:39:16Z", GoVersion:"go1.11.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.5", GitCommit:"753b2dbc622f5cc417845f0ff8a77f539a4213ea", GitTreeState:"clean", BuildDate:"2018-11-26T14:31:35Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): AWS Kops

### PR DESCRIPTION
**Pull request by adshmh**
**What this PR does / why we need it**:
Allow lint command to parse charts with pre-release version. Closes #5150

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: cmd/helm/lint_test.go [hunk 1]
```diff
@@ -20,35 +20,53 @@ import (
...
+			name:      "archived-chart-path",
+			chartPath: "testdata/testcharts/compressedchart-0.1.0.tgz",
+		},
+		{
+			name:      "archived-chart-path-with-hyphens",
+			chartPath: "testdata/testcharts/compressedchart-with-hyphens-0.1.0.tgz",
+		},
+		{
+			name:      "pre-release-chart",
+			chartPath: "testdata/testcharts/pre-release-chart-0.1.0-alpha.tgz",
```
Comment (kivagant-ba) [inline cmd/helm/lint_test.go]: I would suggest adding git hashes that can start from zero or ends with zero plus mixed styles like `0.1.0-0196cc0.1` or `0.1.0-1.0196cc0`

Comment (kivagant-ba) [reply cmd/helm/lint_test.go]: And also suffixes that look like hexadecimal but they aren't. For example, `0.1.0-0x64000`

**Comment by kivagant-ba** [comment]
@thomastaylor312, any chance maintainers can review this? It's a pretty old issue.

**Comment by kivagant-ba** [comment]
Good news, thank you!

File: cmd/helm/lint_test.go (hunk 1 above)
Comment (thomastaylor312) [reply cmd/helm/lint_test.go]: So because of the new logic, I don't think the name of the tarball matters anymore. Am I misreading your feedback?

**Review by thomastaylor312 — APPROVED** [review]
This is looking great! Thank you for refactoring to table unit tests

File: cmd/helm/lint_test.go (hunk 1 above)
Comment (kivagant-ba) [reply cmd/helm/lint_test.go]: I was talking only about negative test cases.

**Pull request merged**